templates/default/activemq.xml.erb

```diff
@@ -110,14 +110,26 @@
         -->
         <transportConnectors>
             <!-- DOS protection, limit concurrent connections to 1000 and frame size to 100MB -->
-            <transportConnector name="openwire" uri="tcp://0.0.0.0:61616?maximumConnections=1000&amp;wireFormat.maxFrameSize=104857600"/>
+            <transportConnector name="openwire" uri="tcp://0.0.0.0:61616?<%=
+              node['activemq']['transport_protocols'] ?
+                "transport.enabledProtocols=#{node['activemq']['transport_protocols']}&amp;" :
+              ''
+              %>maximumConnections=1000&amp;wireFormat.maxFrameSize=104857600"/>
             <transportConnector name="amqp" uri="amqp://0.0.0.0:5672?maximumConnections=1000&amp;wireFormat.maxFrameSize=104857600"/>
 <% if node['activemq']['enable_stomp'] -%>
-            <transportConnector name="stomp" uri="stomp://0.0.0.0:61613?maximumConnections=1000&amp;wireFormat.maxFrameSize=104857600"/>
+            <transportConnector name="stomp" uri="stomp://0.0.0.0:61613?<%= 
+              node['activemq']['transport_protocols'] ?
```

Comment:
Because we need to append to the variable if it is set, and not append if it isn't, we would still need a ternary in this case.